wolfcrypt/src/port/kcapi/kcapi_ecc.c

```diff
@@ -190,14 +192,40 @@ int KcapiEcc_Sign(ecc_key* key, const byte* hash, word32 hashLen, byte* sig,
         }
     }
     if (ret == 0) {
-        ret = kcapi_akcipher_sign(key->handle, hash, hashLen, sig, *sigLen,
+        ret = posix_memalign((void*)&hash_aligned,
```

Comment:
Usually we could only perform the page alloc and memcpy if the incoming hash or signature is not aligned. Maybe add check to allocate only if not `% sysconf(_SC_PAGESIZE) != 0`?

Reply:
Are you able to use the same page allocation for both the hash and signature like you did on the verify? If so then I would keep your always memalign/memcpy logic.

Reply:
The check makes sense.

ECDSA isn't normally supported in libkcapi. The last update was for Linux kernel v4.15. I do not know if the single buffer for verify will work on the sign. I do know the two buffer allocation works for the sign in the case.